examples/dynamic_example/main.tf

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+
+  sub_folders1_var = compact(flatten([for k, i in var.folder_list : i == [] ? [] : [for ip1, op1 in i : join("=1>", [k, ip1])]]))
+
+  sub_folders2_var = compact(flatten([for k, i in var.folder_list : i == [] ? [] : [for ip1, op1 in i : op1 == [] ? [] : [for ip2, op2 in op1 : join("=2>", [join("=1>", [k, ip1]), ip2])]]]))
+
+  sub_folders3_var = compact(flatten([for k, i in var.folder_list : i == [] ? [] : [for ip1, op1 in i : op1 == [] ? [] : [for ip2, op2 in op1 : op2 == [] ? [] : [for ip3, op3 in op2 : join("=3>", [join("=2>", [join("=1>", [k, ip1]), ip2]), ip3])]]]]))
```

Comment:
nit: can we just keep 3 levels and instruct users on how to do more levels if needed?

Reply:
Sure, Will keep till level 3.